src/uploads/swift_lockfile_lock.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash -e
+
+# Source Swift config
+source $(dirname $0)/../configs/swift.public.novarc
+
+set -x
+
+IMAGE_NAME=$1
+# Timeout and sleep time in seconds
+TIMEOUT=${2:-300}
+SLEEP_TIME=5
+
+staging_area=$(mktemp -d)
+
+mkdir -p "${staging_area}/${IMAGE_NAME}"
+
+touch "${staging_area}/${IMAGE_NAME}/lockfile.lock"
+
+pushd "${staging_area}"
+
+# check if the ${IMAGE_NAME}/lockfile.lock exists in the swift container
+# if it does, wait until the timeout is reached and emit an error
+# if it does not, upload the lockfile.lock to the swift container
+# and exit
+while [ $TIMEOUT -gt 0 ]; do
+    swift list $SWIFT_CONTAINER_NAME -p $IMAGE_NAME | grep "lockfile.lock" && sleep $SLEEP_TIME || break
+    TIMEOUT=$(( $TIMEOUT - $SLEEP_TIME ))
+    if [ $TIMEOUT -eq 0 ]; then
```

Comment:
use `$TIMEOUT -lt 1` in case a value that is not multiple of the SLEEP_TIME was used for TIMEOUT.